compress.c

```diff
@@ -1250,7 +1250,10 @@ compress(fdin, fdout)
 
 		ratio = 0;
 		checkpoint = CHECK_GAP;
-		extcode = MAXCODE(n_bits = INIT_BITS)+1;
+		if ((n_bits = INIT_BITS) < maxbits)
+			extcode = MAXCODE(n_bits)+1;
```

Comment:
seems like it'd be a little easier to read if it were:

```c
extcode = MAXCODE(n_bits);
if (n_bits < maxbits)
  extcode += 1
```

wdyt ?

Reply:
Sure, I will update it today, thank you.